test/test_nn.py

```diff
@@ -15548,16 +15547,23 @@ def run_test(*shape):
                 xx = x.cpu().requires_grad_()
             yy = F.log_softmax(xx.float(), dim=-1).to(dtype)
             yy.backward(yy)
-            self.assertEqual(y, yy)
-            self.assertEqual(x.grad, xx.grad)
+            # workaround to reduce memory usage vs. self.assertEqual, see #84944
+            rtol = torch.testing._comparison._DTYPE_PRECISIONS[dtype][0]
+            atol = torch.testing._comparison._DTYPE_PRECISIONS[dtype][1]
+            atol = 0.001 if dtype is torch.half else atol
```

Comment:
We have never exposed it since there was no use case, but internally you can use

https://github.com/pytorch/pytorch/blob/ebd4e90ff7f6d1e57bbf6f2e717a8addb6b28e28/torch/testing/_comparison.py#L84-L86

here:

```suggestion
            rtol, atol = torch.testing._comparison.get_tolerances(dtype, atol=1e-3 if dtype is torch.half else None)
```